test/integration/authorize/list_app_keys_test.rb

```diff
@@ -0,0 +1,132 @@
+require File.expand_path(File.dirname(__FILE__) + '/../../test_helper')
+
+class AuthorizeListAppKeysExtensionTest < Test::Unit::TestCase
+  include TestHelpers::Fixtures
+  include TestHelpers::Integration
+  include TestHelpers::Extensions
+
+  def setup
+    @storage = Storage.instance(true)
+    @storage.flushdb
+
+    Resque.reset!
+    Memoizer.reset!
+
+    setup_provider_fixtures
+
+    @application = Application.save(service_id: @service.id,
+                                    id: next_id,
+                                    state: :active,
+                                    plan_id: @plan_id,
+                                    plan_name: @plan_name)
+
+    @keys = 10.times.map { |i| "key_#{i}" }
+
+    @keys.each do |k|
+      @application.create_key k
+    end
+
+    @application_nokeys = Application.save(service_id: @service.id,
+                                           id: next_id,
+                                           state: :active,
+                                           plan_id: @plan_id,
+                                           plan_name: @plan_name)
+  end
+
+  test 'calling authorize without the list_app_keys extension does not output the application keys section' do
+    get '/transactions/authorize.xml',
+        { provider_key: @provider_key, app_id: @application.id }
+
+    xml_resp = Nokogiri::XML(last_response.body)
+    assert xml_resp.at('app_keys').nil?
```

Comment:
`assert_nil` ?

Reply:
There are other asserts where `assert_nil` could be used.

Reply:
Ok, will take a look!